### DISCUSSION TITLE
Export the returned function/object type of useTranslations hook

### DISCUSSION THREAD

**Discussion by dbrxnds** [amannn/next-intl#128] [Q&A]
Hi! Would it be possible to export the type of the object/function returned by the `useTranslations` hook?

We consume the `useTranslations` hook in a custom `useI18n` hook in our codebase which in turn returns some extra utility stuff required to localize our app. This allows us to have all i18n related stuff in one hook.

We used to just do `type tObj = ReturnType<typeof useTranslations>` but this broke in 2.7.4 as for some reason the type becomes `<NestedKey=never extends NamespaceKeys..` rather than `<NestedKey=any extends NamespaceKeys...` which it actually is.

Any other solution would be greatly appreciated as well. 

Reproduction:
[https://codesandbox.io/s/awesome-bassi-372rsl?file=/src/App.tsx](https://codesandbox.io/s/awesome-bassi-372rsl?file=/src/App.tsx)

I am aware that this reproduction uses a regular React project, but it's just a type issue.

**Comment by amannn** [comment]
Hi @dbrxnds!

Yep, the return type became more complex with the introduction of [typed namespaces and message keys](https://next-intl-docs.vercel.app/docs/usage/typescript).

I'd be curious why you're wrapping the hook in the first place? Is there anything the library could support better?

Practically exporting the return type is a bit difficult and I've chosen not to pursue this further for now: https://github.com/amannn/next-intl/issues/92.

If there's no way for you to not wrap the hook, you might want to cast the return type to a simpler type, but then you can't take advantage of the type safety / autocompletion for namespaces and message keys.

**Reply by dbrxnds** [reply]
Hi, thanks for your quick response!

> I'd be curious why you're wrapping the hook in the first place? Is there anything the library could support better?

Not really, our app just has a couple of strange requirements we need to cover in terms of i18n. This comes with some utility functions that would be a bit confusing to put in it's own hook so we decided to create a single i18n hook which returns all these different things we use throughout our app. We figured it would make more sense to then also re-export the useTranslations (and useIntl) return object from that same hook to make it clear to any dev in our codebase how to handle i18n related things.

> Practically exporting the return type is a bit difficult and I've chosen not to pursue this further for now: https://github.com/amannn/next-intl/issues/92.

My apologies, I looked over this while searching. 

> If there's no way for you to not wrap the hook, you might want to cast the return type to a simpler type, but then you can't take advantage 
of the type safety / autocompletion for namespaces and message keys.

We currently don't use the type safe set up, so I think this will be the solution we end up going with. 

Either way, I appreciate you taking the time to reply!
